### DISCUSSION TITLE
Bug fix

### DISCUSSION THREAD

**Discussion by emiliosaidm** [pay-rails/pay#934] [General]
Hello,

I'm currently working on an app that involves Stripe metered billing. While using the `create_usage_record` method with a specific subscription_item, I encountered an error when executing `subscription.create_usage_record(subscription_item: "si_PKj91eivwl89yw", quantity: 99)`:

```
A parameter provided in the URL (subscription_item) was repeated as a GET or POST parameter. You can only provide this information as a portion of the URL. (Stripe::InvalidRequestError)
```

To resolve this issue, I made adjustments to the `create_usage_record` method in the file `/path/to/your/app/lib/pay/stripe/subscription.rb`. Here's what I did:

- The Pay gem documentation specifies using `subscription_item_id` as the key, but Stripe v9 docs state that `subscription_item` should be used. I updated the key accordingly.

- The `subscription_item` was being sent twice—once as an element of the `options` hash and independently in the method `Stripe::SubscriptionItem.create_usage_record`. According to the Stripe Docs, `subscription_item` should be sent as the first element in the params, followed by a hash with the configuration:

  ```ruby
  Stripe::SubscriptionItem.create_usage_record(
    'si_IutmSSymhsWA5i',
    {
      quantity: 100,
      timestamp: 1571252444,
    },
  )
  ```

  I removed the duplicated `subscription_item` from the hash, resulting in a working solution. The modified `create_usage_record` method looks like this:

  ```ruby
  def create_usage_record(**options)
    subscription_item = options.fetch(:subscription_item, metered_subscription_item&.dig("id"))
    options.delete(:subscription_item)
    ::Stripe::SubscriptionItem.create_usage_record(subscription_item, options, stripe_options)
  end
  ```

I wanted to share this information with you, and I'm open to suggestions if there's a better approach.

Thanks!

**Comment by excid3** [comment]
You are correct, we should remove the `subscription_item_id` from the options hash.
